DEVELOPMENT_GUIDE.md

```diff
@@ -0,0 +1,184 @@
+DEVELOPMENT GUIDE
+=================
+
+**Welcome hacker!**
+
+This document will make your life easier by helping you setup a
+development environment, IDEs, tests, coding practices, or anything that
+will help you be more productive. If you found something is missing or
+inaccurate, update this guide and send a Pull Request.
+
+**Note**: `pyenv` currently only supports macOS and Linux. If you are a
+Windows users, consider using [pipenv](https://docs.pipenv.org/).
+
+1-Click Ready to Hack IDE
+-------------------------
+For setting up a local development environment, we recommend using Gitpod - a service that allows you to spin up an in-browser Visual Studio Code-compatible editor, with everything set up and ready to go for development on this project. Just click the button below to create your private workspace:
+
+[![Gitpod ready-to-code](https://img.shields.io/badge/Gitpod-ready--to--code-blue?logo=gitpod)](https://gitpod.io/#https://github.com/awslabs/aws-sam-cli)
+
+This will start a new Gitpod workspace, and immediately kick off a build of the code. Once it's done, you can start working.
+
+Gitpod is free for 50 hours per month - make sure to stop your workspace when you're done (you can always resume it later, and it won't need to run the build again).
+
+
+Environment Setup
+-----------------
+### 1. Install Python Versions
+
+Our officially supported Python versions are 2.7, 3.6, 3.7 and 3.8. Follow the idioms from this [excellent cheatsheet](http://python-future.org/compatible_idioms.html) to
+make sure your code is compatible with both Python 2.7 and 3 (>=3.6) versions.
+Our CI/CD pipeline is setup to run unit tests against both Python 2.7 and 3 versions. So make sure you test it with both versions before sending a Pull Request.
+See [Unit testing with multiple Python versions](#unit-testing-with-multiple-python-versions).
+
+[pyenv](https://github.com/pyenv/pyenv) is a great tool to
+easily setup multiple Python versions. For 
+
+> Note: For Windows, type
+> `export PATH="/c/Users/<user>/.pyenv/libexec:$PATH"` to add pyenv to
+> your path.
+
+1.  Install PyEnv -
```

Comment:
We'll probably later want to encourage using the Python 3-native `venv`; it's much simpler and requires no ugly hacks:

```shell
python3 -m venv .env
source .env/bin/activate
```

Reply:
+1